src/test/java/io/github/spark_redshift_community/spark/redshift/InMemoryS3AFileSystem.java

```diff
@@ -192,23 +195,29 @@ private boolean isDir(Path f) throws IOException{
 
 
     @Override
-    public  S3AFileStatus getFileStatus(Path f) throws IOException {
+    public S3AFileStatus getFileStatus(Path f) throws IOException {
 
         if (!exists(f)) throw new FileNotFoundException();
+        Tristate state = Tristate.fromBool(dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f)));
 
         if (isDir(f)) {
-            return new S3AFileStatus(
-                true,
-                dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f)),
-                f
+            FileStatus fileStatus = new FileStatus(
+                dataMap.tailMap(toS3Key(f)).size(), true, 1,
+                this.getDefaultBlockSize(), System.currentTimeMillis(), f
+            );
+            return S3AFileStatus.fromFileStatus(
+                fileStatus, Tristate.fromBool(
+                    dataMap.tailMap(toS3Key(f)).size() == 1 && dataMap.containsKey(toS3Key(f))
+                )
             );
         }
         else {
             return new S3AFileStatus(
                 dataMap.get(toS3Key(f)).toByteArray().length,
                 System.currentTimeMillis(),
                 f,
-                this.getDefaultBlockSize()
+                this.getDefaultBlockSize(),
+                "owner"
```

Comment:
why do we need to add "owner" here? This looks like a magic string, would be nice to have a small comment explaining this

Reply:
It's a test case with a test value. The constructor definition in hadoop 3.2.1 requires this.